packages/core/src/tools/utils.ts

```diff
@@ -244,6 +244,10 @@ export function mapValues<A, B>(object: { [key: string]: A }, fn: (arg: A) => B)
   return newObject
 }
 
+export function notEmpty<T>(value: T | null | undefined): value is T {
```

Comment:
Empty seems more suited to a collection type, from you implementation [nullish](https://developer.mozilla.org/en-US/docs/Glossary/Nullish) could be a more suited term